### PR TITLE
DOC: Capitalize docstring summaries of Series

### LINKED ISSUE #23641: DOC: Capitalize docstring summaries of Series
Our standard is to start the docstrings with a capital letter (and finish them with a period).

We'll shortly add a check to make sure that new contributions follow this standard. But before we can do that, we need to fix the many cases that are already wrong.

In this issue we should fix all the cases of methods of Series. This is the command to generate them (which should return nothing after the fix), and the list of current issues.
```
$ ./scripts/validate_docstrings.py --prefix=pandas.Series --errors=SS02
pandas.Series.dtype: Summary does not start with a capital letter
pandas.Series.ftype: Summary does not start with a capital letter
pandas.Series.shape: Summary does not start with a capital letter
pandas.Series.nbytes: Summary does not start with a capital letter
pandas.Series.ndim: Summary does not start with a capital letter
pandas.Series.size: Summary does not start with a capital letter
pandas.Series.strides: Summary does not start with a capital letter
pandas.Series.itemsize: Summary does not start with a capital letter
pandas.Series.base: Summary does not start with a capital letter
pandas.Series.T: Summary does not start with a capital letter
pandas.Series.hasnans: Summary does not start with a capital letter
pandas.Series.flags: Summary does not start with a capital letter
pandas.Series.dtypes: Summary does not start with a capital letter
pandas.Series.ftypes: Summary does not start with a capital letter
pandas.Series.data: Summary does not start with a capital letter
pandas.Series.get_values: Summary does not start with a capital letter
pandas.Series.item: Summary does not start with a capital letter
pandas.Series.argmin: Summary does not start with a capital letter
pandas.Series.argmax: Summary does not start with a capital letter
pandas.Series.dt.round: Summary does not start with a capital letter
pandas.Series.dt.floor: Summary does not start with a capital letter
pandas.Series.dt.ceil: Summary does not start with a capital letter
pandas.Series.to_msgpack: Summary does not start with a capital letter
pandas.Series.sparse: Summary does not start with a capital letter
pandas.Series.transpose: Summary does not start with a capital letter
```

In some cases the problem is that there is a deprecated directive `.. deprecated::` at the beginning of te docstring. This is wrong too, and the deprecation should be after the short summary (the first line/paragraph of the summary).

While this issue is mainly to make the first letter in the summary a capital letter, for the cases where this is fix, would be nice to also add a period at the end of the paragraph, if there is not one already.

### PR DESCRIPTION
**Pull request by avolkov**
Updated documentation to resolve SS02 errors in pandas.Series

- [x] closes #23641
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by avolkov** [comment]
@datapythonista

Sure, I've noticed some of these strings were inconsistent.

I'll update current PR with fixes.

File: pandas/core/base.py [hunk 1]
```diff
@@ -715,50 +719,62 @@ def item(self):
...
+        """
+        Return the number of elements in the underlying data.
+        """
         return self._values.size
 
     @property
     def flags(self):
-        """ return the ndarray.flags for the underlying data """
+        """
+        Return the ndarray.flags for the underlying data
```
Comment (datapythonista) [inline pandas/core/base.py]:
```suggestion
        Return the ndarray.flags for the underlying data.
```

File: pandas/core/base.py [hunk 2]
```diff
@@ -715,50 +719,62 @@ def item(self):
...
                       "in a future version".format(obj=type(self).__name__),
                       FutureWarning, stacklevel=2)
         return self.values.flags
 
     @property
     def base(self):
-        """ return the base object if the memory of the underlying data is
-        shared
+        """
+        Return the base object if the memory of the underlying data is shared
```
Comment (datapythonista) [inline pandas/core/base.py]:
```suggestion
        Return the base object if the memory of the underlying data is shared.
```

File: pandas/core/generic.py [hunk 3]
```diff
@@ -2328,7 +2328,7 @@ def to_hdf(self, path_or_buf, key, **kwargs):
 
     def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         """
-        msgpack (serialize) object to input file path
+        Serialize object to input file path using msgpack format
```
Comment (datapythonista) [inline pandas/core/generic.py]:
```suggestion
        Serialize object to input file path using msgpack format.
```

File: pandas/core/base.py [hunk 4]
```diff
@@ -668,11 +668,13 @@ class IndexOpsMixin(object):
...
-        """ return the transpose, which is by definition self """
+        """
+        Return the transpose, which is by definition self.
+        """
         nv.validate_transpose(args, kwargs)
         return self
 
-    T = property(transpose, doc="return the transpose, which is by "
+    T = property(transpose, doc="Return the transpose, which is by "
                                 "definition self")
```
Comment (datapythonista) [inline pandas/core/base.py]:
```suggestion
                                "definition self.")
```

**Review by datapythonista — APPROVED** [review]
Thanks @avolkov much better now. I added few missing periods that you should be able to accept just with a click. Happy to merge after that.

**Comment by datapythonista** [comment]
Still no luck with the CI, can you merge master again please

**Comment by avolkov** [comment]
I was sure something was wrong in master. If that doesn't work and re-run tests locally cherry-picking the changes.

**Comment by datapythonista** [comment]
@avolkov just realized that we have a test that it's checking the text in a docstring. That's what is making the CI fail. Can you take a look and fix it please?

This is what it's failing I think: `pandas/tests/series/test_internals.py:320`
```
    def test_hasnans_unchached_for_series():
        # GH#19700
        idx = pd.Index([0, 1])
        assert idx.hasnans is False
        assert 'hasnans' in idx._cache
        ser = idx.to_series()
        assert ser.hasnans is False
        assert not hasattr(ser, '_cache')
        ser.iloc[-1] = np.nan
        assert ser.hasnans is True
>       assert Series.hasnans.__doc__ == pd.Index.hasnans.__doc__
E       AssertionError: assert '\n        Re...ps.\n        ' == ' return if I ...erf speedups '
E         - 
E         -         Return if I have any nans; enables various perf speedups.
E         ?  ^^^^^^^^                                                       ^^
E         +  return if I have any nans; enables various perf speedups 
E         ?  ^                                                       ^
E         -
```

**Comment by datapythonista** [comment]
@avolkov can you fix the error in travis please:
```
./pandas/core/base.py:705:80: E501 line too long (80 > 79 characters)
```

File: pandas/core/base.py [hunk 5]
```diff
@@ -702,7 +702,8 @@ def shape(self):
     @property
     def ndim(self):
         """
-        Return the number of dimensions of the underlying data, by definition 1.
+        Return the number of dimensions of the underlying data,
+        by definition 1.
```
Comment (datapythonista) [inline pandas/core/base.py]: Sorry about that @avolkov, but we'll validate soon that the first paragraph in the docstrings is a single line (it's shown in the summaries, and if it's too long the format gets wrong and things are difficult to read).

Do you mind removng the `Return the`, and start in the `Number of dimensions...` so this fits in a single line?

**Review by datapythonista — APPROVED** [review]
Looks great, thank you @avolkov

**Comment by jreback** [comment]
thanks!

**Pull request merged**